src/ripple/app/tests/JTx.h

```diff
@@ -237,48 +300,442 @@ class fee
+    {
+    }
+
+    void
+    operator()(Env const&, JTx& jt) const;
+};
+
+/** Set the expected result code for a JTx
+    The test will fail if the code doesn't match.
+*/
+class ter
+{
+private:
+    TER v_;
+
+public:
+    explicit
+    ter (TER v)
+        : v_(v)
+    {
+    }
+
+    void
+    operator()(Env const&, JTx& jt) const
+    {
+        jt.ter = v_;
+    }
+};
+
+//------------------------------------------------------------------------------
+//
+// Conditions
+//
+//------------------------------------------------------------------------------
+
+namespace detail {
+
+inline
+void
+require_args (requires_t& vec)
+{
+}
+
+template <class Cond, class... Args>
+inline
+void
+require_args (requires_t& vec,
+    Cond const& cond, Args const&... args)
+{
+    vec.push_back(cond);
+    require_args(vec, args...);
+}
+
+} // detail
+
+// Standalone function composes
+// one condition functor from many.
+template <class...Args>
+detail::require_t
+required (Args const&... args)
+{
+    detail::requires_t vec;
+    detail::require_args(vec, args...);
+    return [vec](Env const& env)
+    {
+        for(auto const& f : vec)
+            f(env);
+    };
+}
+
+/** Check a set of conditions.
+
+    The conditions are checked after a JTx is
+    applied, and only if the resulting TER
+    matches the expected TER.
+*/
+class require
+{
+private:
+    detail::require_t cond_;
+
+public:
+    template<class... Args>
+    require(Args const&... args)
+        : cond_(required(args...))
+    {
+    }
+
+    void
+    operator()(Env const&, JTx& jt) const
+    {
+        jt.requires.emplace_back(cond_);
+    }
+};
+
+//
+// Conditions
+//
+
+namespace cond {
+
+/** A balance matches.
+
+    This allows "none" which means either the account
+    doesn't exist (no XRP) or the trust line does not
+    exist. If an amount is specified, the SLE must
+    exist even if the amount is 0, or else the test
+    fails.
+*/
+class balance
+{
+private:
+    bool none_;
+    Account account_;
+    STAmount value_;
+
+public:
+    balance (Account const& account, none_t,
+            Issue const& issue = XRP)
+        : account_(account)
+        , none_(true)
+        , value_(issue)
+    {
+    }
+
+    balance (Account const& account,
+            STAmount const& value)
+        : none_(false)
+        , account_(account)
+        , value_(value)
+    {
+    }
+
+    void
+    operator()(Env const&) const;
+};
+
+namespace detail {
+
+class flags_helper
+{
+protected:
+    std::uint32_t mask_;
+
+private:
+    inline
+    void
+    set_args()
+    {
+    }
+
+    void
+    set_args (std::uint32_t flag)
+    {
+        switch(flag)
+        {
+        case asfRequireDest:    mask_ |= lsfRequireDestTag; break;
+        case asfRequireAuth:    mask_ |= lsfRequireAuth; break;
+        case asfDisallowXRP:    mask_ |= lsfDisallowXRP; break;
+        case asfDisableMaster:  mask_ |= lsfDisableMaster; break;
+        //case asfAccountTxnID: // ???
```

Comment:
It can currently only be set by using `SetFlag`

```c++
        // Track transaction IDs signed by this account in its root
        //
        if ((uSetFlag == asfAccountTxnID) && !mTxnAccount->isFieldPresent (sfAccountTxnID))
        {
            m_journal.trace << "Set AccountTxnID";
            mTxnAccount->makeFieldPresent (sfAccountTxnID);
         }

        if ((uClearFlag == asfAccountTxnID) && mTxnAccount->isFieldPresent (sfAccountTxnID))
        {
            m_journal.trace << "Clear AccountTxnID";
            mTxnAccount->makeFieldAbsent (sfAccountTxnID);
        }
```